File: src/main/java/burp/model/InputValueMap.java
package burp.model;

import org.apache.jena.rdf.model.Resource;

import burp.vocabularies.RML;

import java.util.List;

public class InputValueMap extends TermMap {
	
	public DatatypeMap datatypeMap = null;
	public LanguageMap languageMap = null;
	
	public InputValueMap() {
		termType = RML.LITERAL;
	}

	@Override
	public String getName() {
		return "input value map";
	}

	@Override
	public List<Resource> getAllowedTermTypes() {
		return List.of(RML.IRI, RML.URI, RML.BLANKNODE, RML.LITERAL);
	}

	@Override
	public boolean isGatherMap() {
		return false;
	}
	
}

File: src/main/java/burp/model/InputValueMap.kt
package burp.model

import burp.vocabularies.RML

class InputValueMap : TermMap() {

    init {
        termType = RML.LITERAL
    }

    override fun getName(): String = "input value map"

    override fun getAllowedTermTypes()=listOf(RML.IRI, RML.URI, RML.BLANKNODE, RML.LITERAL)

    override fun isGatherMap() = false
}